converters/toZigbee.js

```diff
@@ -1429,6 +1429,74 @@ const converters = {
             await entity.read('closuresDoorLock', ['lockState']);
         },
     },
+    pincode_lock: {
+        key: ['pin_code'],
+        convertSet: async (entity, key, value, meta) => {
+            const user = value.user;
+            const pinCode = value.pin_code;
+            if ( isNaN(user) ) {
+                throw new Error('user must be numbers');
+            }
+            if ( !converters.pincode_lock.userInRange(meta, user) ) {
+                throw new Error('user must be in range for device');
+            }
+            if (pinCode === undefined || pinCode === null) {
+                await entity.command(
+                    'closuresDoorLock',
+                    'clearPinCode',
+                    {
+                        'userid': user,
+                    },
+                    getOptions(meta.mapped),
+                );
+            } else {
+                if (isNaN(pinCode)) {
+                    throw new Error('pinCode must be a number or pinCode');
+                }
+                await entity.command(
+                    'closuresDoorLock',
+                    'setPinCode',
+                    {
+                        'userid': user,
+                        'userstatus': 1,
+                        'usertype': 0,
+                        'pincodevalue': pinCode.toString(),
+                    },
+                    getOptions(meta.mapped),
+                );
+            }
+            return {readAfterWriteTime: 200};
+        },
+        convertGet: async (entity, key, meta) => {
+            const user = meta && meta.message && meta.message.pin_code ? meta.message.pin_code.user : undefined;
+            if (user === undefined) {
+                const max = meta.mapped.meta.pinCodeCount;
+                // Get all
+                for (let i = 0; i < max; i++) {
+                    await converters.pincode_lock.getPinCode(entity, i, meta);
+                }
+            } else {
+                if (isNaN(user)) {
+                    throw new Error('user must be numbers');
+                }
+                if (!converters.pincode_lock.userInRange(meta, user)) {
+                    throw new Error('userId must be in range for device');
+                }
+                await converters.pincode_lock.getPinCode(entity, user, meta);
+            }
+        },
+        getPinCode: async (entity, user, meta) => {
+            await entity.command(
+                'closuresDoorLock',
+                'getPinCode',
+                {
+                    'userid': user,
+                },
+                getOptions(meta.mapped),
+            );
+        },
+        userInRange: (meta, user) => user >= 0 && user < meta.mapped.meta.pinCodeCount,
```

Comment:
I prefer adding a new method to https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/utils.js 

```js
function isInRange(min, max, value) {
   return value >= min && value =< max
}
```

and call it with `utils.isInRange(0, meta.mapped.meta.pinCodeCount -1, user)`

We don't want to expose any other methods beside `convertSet` and `convertGet` here.